mrtt-ui/src/components/TabularInput/TabularInputRow.js

```diff
@@ -0,0 +1,55 @@
+import { useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+import { TextField } from '@mui/material'
+import { Delete, Save } from '@mui/icons-material'
+import { styled } from '@mui/material/styles'
+
+import { TabularInputSection, TabularLabel } from '../../styles/forms'
+
+const TabularInputRow = ({ label, value, index, deleteMeasurementItem, updateMeasurementItem }) => {
+  const [initialVal, setInitialVal] = useState('')
+  const [currentVal, setCurrentVal] = useState('')
+  const handleDelete = () => {
+    deleteMeasurementItem(index)
+  }
+
+  const handleUpdate = () => {
+    updateMeasurementItem(index, currentVal)
+  }
+
+  useEffect(() => {
+    if (value) {
+      setCurrentVal(value)
+      setInitialVal(value)
+    }
+  }, [value])
```

Comment:
can we name this effect after whatever its doing?

Reply:
will do!